bitnami/openldap/2.6/debian-12/rootfs/opt/bitnami/scripts/openldap/run.sh

```diff
@@ -21,14 +21,60 @@ command="$(command -v slapd)"
 # https://github.com/docker/docker/issues/8231
 ulimit -n "$LDAP_ULIMIT_NOFILES"
 
-flags=("-h" "ldap://:${LDAP_PORT_NUMBER}/ ldapi:///")
+declare -a flags
+declare -A flags_map
+
+# Drop privileges if we start as root
+am_i_root && flags_map["-u"]="${LDAP_DAEMON_USER}"
+
+# Set config dir
+flags_map["-F"]="${LDAP_CONF_DIR}/slapd.d"
+
+# Enable debug with desired level
+flags_map["-d"]="${LDAP_LOGLEVEL}"
+
+# The LDAP IPC is always on
+flags_map["-h"]+="${flags_map["-h"]:+" "}ldapi:///"
```

Comment:
Why not just simply this?

```suggestion
flags_map["-h"]="ldapi:///"
```

`flags_map["-h"]` is being declared here for the first time, so the value is going to be always `ldapi:///` right? And I think it's easier to read and understand.

Reply:
This is just stable idiom to insert arg in multi-valued flag, which in turn give possibility to move/add/remove additional "blocks" between declare and final assembling without affecting overall flow.
Idea that is:
- single-valued flags sets as `flags_map["<flag>"]="<arg>"`, so the last setter wins.
- multi-valued flags sets as `flags_map["<flag>"]+="${flags_map["<flag>"]:+"<delim>"}<arg>"`, so args line compiled natively and "block" position only affect arg sorting for particular flag.

Reply:
I see your point 👍